commands/pipelines/promote.js

```diff
@@ -55,21 +58,60 @@ module.exports = {
       }));
 
     const sourceStage = coupling.stage;
-    const targetStage = PROMOTION_ORDER[PROMOTION_ORDER.indexOf(sourceStage) + 1];
 
-    if (targetStage === null || PROMOTION_ORDER.indexOf(sourceStage) < 0) {
-      throw new Error(`Cannot promote ${app} from '${sourceStage}' stage`);
-    }
+    let promotionActionName = '';
+    let targetApps = []
+    if (context.flags.to){
+      // The user specified a specific set of apps they want to target
+      // We don't have to infer the apps or the stage they want to promote to
+
+      let targetAppNames = context.flags.to.split(',').filter((appName)=>{
+        // Strip out any empty app names due to something like a trailing comma
+        return appName.length >= 1;
+      });
+
+      // Now let's make sure that we can find every target app they specified
+      // The only requirement is that the app be in this pipeline. They can be at any stage.
+      targetApps = targetAppNames.map((targetAppName)=>{
+        console.log(`target app name: ${targetAppName} and source app name ${app}`);
```

Comment:
I think this may be a left over debugging statement? I think we should consider removing it or at least rewording and using `cli.log`.